plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -4251,6 +4257,34 @@ public void rebootVM(final Connection conn, final VM vm, final String vmName) th
         }
     }
 
+    protected String skipOrRemoveSR(Connection conn, SR sr) {
```

Comment:
@anshul1886 @DaanHoogland @rodrigo93 As an ideia. Maybe it could have a name like "removeSrIfNotNull" or "removeSrIfVdiOperationAndSrNotNull".

As an example, a documetation could look like:
`/** Removes the SR if the SR is not null and exists a VDI operation for this SR*/`

Do you think that one of this options (or anoter in your mind) could improve this code? :-)

Thanks